mod/execution/pkg/deposit/sync.go

```diff
@@ -37,6 +37,15 @@ func (s *Service[
 	BeaconBlockT, _, _, _, _,
 ]) depositFetcher(ctx context.Context, event async.Event[BeaconBlockT]) {
 	blockNum := event.Data().GetBody().GetExecutionPayload().GetNumber()
+	if blockNum < s.eth1FollowDistance {
```

Comment:
correct me If I'm wrong, but just confirming, this case is only triggered when blockNum is 0?

Reply:
as the values are set right now yes. However that is not really hit, since we don't emit a finalized block event on genesis

Reply:
eth1FollowDistance 
```suggestion
	if blockNum < s.eth1FollowDistance + 1 {
```

Will never need to query the logs from block height 0. So the first block to query for should be 1.